python/tank/context.py

```diff
@@ -579,8 +579,8 @@ def _fields_from_entity_paths(self, template):
             
             # walk up path until we reach the project root and get values
             while cur_path not in project_roots:
-                if template.validate(cur_path):
-                    cur_fields = template.get_fields(cur_path)
+                cur_fields = template.validate_and_get_fields(cur_path)
+                if cur_fields != None:
```

Comment:
more standard to use `if cur_fields is not None` ( see http://stackoverflow.com/questions/3965104/not-none-test-in-python and https://www.python.org/dev/peps/pep-0008/#programming-recommendations )